### ISSUE TITLE
SQL NULLIF function doesn't work in numeric expressions

### ISSUE THREAD

**Issue by msiomkin** [tarantool/tarantool#6989]
**Bug description**

subj

* OS: Linux
* OS Version: Centos 8
* Architecture: amd64

Tarantool 2.10.0-beta2-207-ga5557281c
Target: Linux-x86_64-Release
Build options: cmake . -DCMAKE_INSTALL_PREFIX=/usr/local -DENABLE_BACKTRACE=ON
Compiler: /usr/lib64/ccache/cc /usr/lib64/ccache/c++
C_FLAGS: -fexceptions -funwind-tables -fno-omit-frame-pointer -fno-stack-protector -fno-common -fopenmp -msse2 -std=c11 -Wall -Wextra -Wno-strict-aliasing -Wno-char-subscripts -Wno-format-truncation -Wno-gnu-alignof-expression -fno-gnu89-inline -Wno-cast-function-type
CXX_FLAGS: -fexceptions -funwind-tables -fno-omit-frame-pointer -fno-stack-protector -fno-common -fopenmp -msse2 -std=c++11 -Wall -Wextra -Wno-strict-aliasing -Wno-char-subscripts -Wno-format-truncation -Wno-invalid-offsetof -Wno-gnu-alignof-expression -Wno-cast-function-type

**Steps to reproduce**

Run tarantool, box.cfg{}

Run query:
```lua
box.execute([[select 1 / nullif(1, 0)]])
```

**Actual behavior**

```
---
- null
- 'Type mismatch: can not convert scalar(1) to integer, decimal or double'
...
```

**Expected behavior**

```
---
- metadata:
  - name: COLUMN_1
    type: integer
  rows:
  - [1]
...
```

**Comment by ImeevMA** [comment]
Now `NULLIF()` returns SCALAR. I believe @msiomkin wants it to return the same type as the first argument (no matter if it will be the first argument or NULL). @unera what do you think?

**Comment by msiomkin** [comment]
Implicit casting (if possible or an error otherwise) of all branches to the type of the first branch result sounds reasonably.

**Comment by msiomkin** [comment]
Returning the first argument type sounds reasonably. If the second argument may be implicitly converted to the first argument type it should be probably done.

**Comment by unera** [comment]
I think `NULLIF` must return type of the first argument.

```
NULLIF(INTEGER, *) -> returns NULL::INTEGER or the first INTEGER
```

**Comment by ImeevMA** [comment]
Will be fixed in #6990.
